src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,216 @@
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                    "database/categories_books/delete-data-from-categories-books-table.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                    "database/books/delete-books-from-books-table.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                        "database/categories/delete-categories-from-table-categories.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new category")
+    void createCategory_ValidRequestDto_Success() throws Exception {
+        CategoryRequestDto requestDto = CategoryProvider.createCategoryRequestDto();
+
+        CategoryResponseDto expected = new CategoryResponseDto()
+                .setName(requestDto.getName())
+                .setDescription(requestDto.getDescription());
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(post("/categories")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+            )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        CategoryResponseDto actual =
+                objectMapper.readValue(result.getResponse().getContentAsString(),
+                CategoryResponseDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual);
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Get a category by id")
+    void getCategoryById_ValidCategoryId_ShouldReturnCategoryResponseDto() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories/{id}", TEST_CATEGORY_ID)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryResponseDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), CategoryResponseDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(TEST_CATEGORY_ID, actual.getId());
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Get books list by category id")
+    @Sql(scripts = "classpath:database/categories_books/add-books-to-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/categories_books"
+                + "/delete-data-from-categories-books-table.sql",
```

Comment:
```suggestion
    @Sql(scripts = "classpath:database/categories_books/clear-categories-books-table.sql",
```
let's try to fit it in one line